src/diffusers/models/modeling_utils.py

```diff
@@ -123,6 +128,111 @@ def _determine_device_map(model: "ModelMixin", device_map, max_memory, torch_dty
     return device_map
 
 
+# Taken from https://github.com/huggingface/transformers/blob/main/src/transformers/modeling_utils.py
```

Comment:
Maybe we can even make use of @Wauplin work on this [PR](https://github.com/huggingface/huggingface_hub/pull/1938) ? Right now, transformers, peft and accelerate have their own utility methods for sharding. So maybe we should all shift to the one on `huggingface_hub` ?

Reply:
We can reuse but `transformers` is NOT a required dependency. So, users should be able to initialize a `diffusers` model (such as `Transformer2D`) without having to install `transformers`. 

@SunMarc could be doable but I would like to be a little bit more stabilized first because `transformers` and `diffusers` have differences in terms of how they initialize the model (the former directly passes a `config` object while the latter one doesn't).

Reply:
> We can reuse but `transformers` is NOT a required dependency

I see, thanks for the info.